File: iOS/SayHi/SayHi/StageCategory.swift
//
//  StageCategory.swift
//  SayHi
//
//  Created by Klemenz, Oliver on 23.12.16.
//  Copyright © 2016 Klemenz, Oliver. All rights reserved.
//

import UIKit
import Foundation

struct StageCategory {
    var key: String = ""
    var langCode: String = ""
    var name: String = ""
    var primaryLangKey: String = ""
    var counter: Int = 1
    
    var category: Category {
        let category = Category(key: key,
                                langCode: langCode,
                                name: name,
                                color: CategoryStagedColor,
                                icon: CategoryStagedIcon,
                                order: 0,
                                primaryLangKey: primaryLangKey)
        category.stage = true
        return category
    }
}

File: Android/SayHi/app/src/main/java/de/oklemenz/sayhi/model/StageCategory.java
package de.oklemenz.sayhi.model;

import java.io.Serializable;

/**
 * Created by Oliver Klemenz on 22.02.17.
 */

public class StageCategory implements Serializable {

    public String key = "";
    public String langCode = "";
    public String name = "";
    public String primaryLangKey = "";
    public Long counter = 1L;

    public StageCategory(String key, String langCode, String name, String primaryLangKey, Long counter) {
        this.key = key;
        this.langCode = langCode;
        this.name = name;
        this.primaryLangKey = primaryLangKey;
        this.counter = counter;
    }

    public Category category() {
        Category category = new Category(
                key,
                langCode,
                name,
                Category.CategoryStagedColor,
                Category.CategoryStagedIcon,
                0,
                primaryLangKey);
        category.stage = true;
        return category;
    }
}
